ddtrace/debugging/_capture/tracing.py

```diff
@@ -62,3 +67,61 @@ def exit(self, retval, exc_info, duration):
 
     def line(self):
         raise NotImplementedError("Dynamic line spans are not supported in Python")
+
+
+@attr.s
+class SpanDecoration(CapturedEvent):
+    """Decorate a span."""
+
+    def _decorate_span(self, _locals):
+        # type: (t.Dict[str, t.Any]) -> None
+        probe = t.cast(SpanDecorationMixin, self.probe)
+
+        if probe.target_span == SpanDecorationTargetSpan.ACTIVE:
+            span = ddtrace.tracer.current_span()
+        elif probe.target_span == SpanDecorationTargetSpan.ROOT:
+            span = ddtrace.tracer.current_root_span()
+        else:
+            log.error("Invalid target span for span decoration: %s", probe.target_span)
+            return
+
+        if span is not None:
+            log.debug("Decorating span %r according to span decoration probe %r", span, probe)
+            for d in probe.decorations:
+                if not (d.when is None or d.when(_locals)):
+                    continue
+                for tag in d.tags:
+                    span.set_tag(tag.name, tag.value(_locals))
```

Comment:
what happen if that fails and raise an error?

Reply:
good point!